src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -68,6 +70,28 @@ public static class TagHelperBlockRewriter
                 // Only want to track the attribute if we succeeded in parsing its corresponding Block/Span.
                 if (succeeded)
                 {
+                    string attributeValueType;
+
+                    // Check if it's a bound attribute that is not of type string. If there's an empty value we need 
+                    // to create an error.
+                    if (attributeValueTypes.TryGetValue(attribute.Key, out attributeValueType) &&
+                       !IsStringAttribute(attributeValueType) &&
+                       (attribute.Value == null ||
+                       (!attribute.Value.IsBlock && string.IsNullOrEmpty(((Span)attribute.Value).Content)) ||
+                       (attribute.Value.IsBlock && ((Block)attribute.Value).Children.Count() == 0)))
+                    {
+                        var errorLocation = GetAttributeNameStartLocation(child);
```

Comment:
Sadly it does not. The node contains leading whitespace which can be contained in a `Block`. Hence we need to find the first symbol that is not whitespace leading up to the attribute name. To do that we have to dive through blocks to find it :cry: